initfiles/componentfiles/configschema/xsd/esp.xsd

```diff
@@ -189,7 +189,7 @@
                             </xs:complexType>
                         </xs:element>
 
-                        <xs:element name="Instance" hpcc:itemType="hwinstance" maxOccurs="unbounded" hpcc:class="valueSet">
+                        <xs:element name="Instance" hpcc:itemType="hwinstance" maxOccurs="unbounded" hpcc:class="valueSet" hpcc:requiredInstanceComponents="/Evnironment/Software/[#itemtype='dafilesrv']">
```

Comment:
The '#' is a special extension of XPaths in the config manager. The # is used to select environment nodes based on schema values, not environment values. For example, it can be used to select all ECL watch esp services. There is no way to do that using the environment because there is no unique XML signature (all esp services are EspService elements).